include/volume/volume_cooling_nonspherical_gaussians_crhmc.hpp

```diff
@@ -0,0 +1,612 @@
+        
+        CrhmcProblem problem = CrhmcProblem(input);
+
+       if(problem.terminate) { return; }
+
+        problem.options.simdLen = simdLen;
+
+        crhmc_walk_params params(input.df, start_point.dimension(), problem.options);
+
+        //eta initialization with the previous walk eta
+        problem.options.etaInitialize = false;
+        params.eta = initial_eta;
+
+        // Create the walk object for this problem
+        CRHMCWalkType walk = CRHMCWalkType(problem, start_point, input.df, input.f, params);
+
+        // Compute some ratios to decide if this is the last gaussian
+        for (unsigned int j = 0; j < steps; j++) 
+        {
+            walk.apply(rng, walk_length);
+            start_point = walk.getPoint();
+            curr_its += 1.0;    
+            curr_fn += eval_exp(start_point, inv_covariance_matrix, next_a, a_vals[it]);
+        }
+
+        //restore the new eta and start point, by looking at the walk after its operations
+        initial_eta = walk.get_current_eta();
+        //start_point = walk.getPoint();        
+
+        // Remove the last gaussian.
+        // Set the last a_i equal to zero
+        if (next_a > 0 && curr_fn / curr_its > (1.0 + tol)) 
+        {
+            a_vals.push_back(next_a);
+            it++;
+        } else if (next_a <= 0) 
+        {
+            a_vals.push_back(a_stop);
+            it++;
+            break;
+        } else {
+            a_vals[it] = a_stop;
+            break;
+        }
+    }
+}
+
+
+template <typename NT>
+struct non_gaussian_annealing_parameters
+{
+    non_gaussian_annealing_parameters(unsigned int d)
+        : frac(0.1)
+        , ratio(NT(1) - NT(1) / NT(d))
+        , C(NT(2))
+        , N(500 * ((int) C) + ((int) (d * d)))
+        , W(6 * d * d + 800)
+    {}
+
+    NT frac;
+    NT ratio;
+    NT C;
+    unsigned int N;
+    unsigned int W;
+};
+
+
+template
+<
+    typename Polytope,
+    typename RandomNumberGenerator,
+    typename WalkTypePolicy = CRHMCWalk,
+    int simdLen = 8
+>
+double non_spherical_crhmc_volume_cooling_gaussians(Polytope& Pin,
+                                RandomNumberGenerator& rng,
+                                double const& error = 0.1,
+                                unsigned int const& walk_length = 1)
+{
+    typedef typename Polytope::PointType Point;
+    typedef typename Point::FT     NT;
+    typedef typename Polytope::VT  VT;
+    typedef typename Polytope::MT  MT;
+    typedef typename NonSphericalGaussianFunctor::FunctionFunctor<Point>    Func;
+    typedef typename NonSphericalGaussianFunctor::GradientFunctor<Point>    Grad;
+    typedef typename NonSphericalGaussianFunctor::HessianFunctor<Point>     Hess;
+    typedef typename NonSphericalGaussianFunctor::parameters<NT, Point>     func_params;
+
+    typedef crhmc_input<MT, Point, Func, Grad, ZeroFunctor<Point>> Input;
+    //typedef crhmc_input<MT, Point, Func, Grad, Hess> Input;   
```

Comment:
I'd remove everything related to the Hessian, we should use the zero functor only.